File: Factory/AbstractFactory/Java/src/PizzaStore/CaliforniaPizzaStore.java
package PizzaStore;

import IngredientFactory.CaliforniaPizzaIngredientFactory;
import IngredientFactory.PizzaIngredientFactory;
import Pizza.Pizza;
import Pizza.CheesePizza;
import Pizza.ClamPizza;
import Pizza.PepperoniPizza;
import Pizza.VeggiePizza;

public class CaliforniaPizzaStore extends PizzaStore {
    @Override
    protected Pizza createPizza(String type) {
        Pizza pizza = null;
        PizzaIngredientFactory pizzaIngredientFactory = new CaliforniaPizzaIngredientFactory();
        if(type.equals("cheese")){
            pizza = new CheesePizza(pizzaIngredientFactory);
            pizza.setName("California Style Cheese Pizza");
        }

        if(type.equals("veggie")){
            pizza = new VeggiePizza(pizzaIngredientFactory);
            pizza.setName("California Style Veggie Pizza");
        }

        if(type.equals("clam")){
            pizza = new ClamPizza(pizzaIngredientFactory);
            pizza.setName("California Style Clam Pizza");
        }

        if(type.equals("pepperoni")){
            pizza = new PepperoniPizza(pizzaIngredientFactory);
            pizza.setName("California Style Pepperoni Pizza");
        }
        return pizza;
    }
}

File: Factory/AbstractFactory/TypeScript/src/PizzaStore/CaliforniaPizzaStore.ts
import Pizza from '../Pizza/Pizza';
import CheesePizza from '../Pizza/CheesePizza';
import PepperoniPizza from '../Pizza/PepperoniPizza';
import ClamPizza from '../Pizza/ClamPizza';
import VeggiePizza from '../Pizza/VeggiePizza';
import PizzaStore from './PizzaStore';
import PizzaIngredientFactory from '../IngredientFactory/PizzaIngredientFactory';
import CaliforniaPizzaIngredientFactory from '../IngredientFactory/CaliforniaPizzaIngredientFactory';

export default class CaliforniaPizzaStore extends PizzaStore {
  protected createPizza(type: string): Pizza {
    let pizza: Pizza = null;
    const pizzaIngredientFactory: PizzaIngredientFactory =
      new CaliforniaPizzaIngredientFactory();

    if (type === 'cheese') {
      pizza = new CheesePizza(pizzaIngredientFactory);
      pizza.setName('California Style Cheese Pizza');
    }

    if (type === 'veggie') {
      pizza = new VeggiePizza(pizzaIngredientFactory);
      pizza.setName('California Style Veggie Pizza');
    }

    if (type === 'pepperoni') {
      pizza = new PepperoniPizza(pizzaIngredientFactory);
      pizza.setName('California Style Pepperoni Pizza');
    }

    if (type === 'clam') {
      pizza = new ClamPizza(pizzaIngredientFactory);
      pizza.setName('California Style Clam Pizza');
    }

    return pizza;
  }
}
